### ISSUE TITLE
Need to clarify what should be in profile crate for additional terms

### ISSUE THREAD

**Issue by ptsefton** [ResearchObject/ro-crate#258]
**Describe the bug**

The profile crate does not work the way I was expecting (may not be a bug, but I want to clarify this)

in the [crate-focus section](https://github.com/ResearchObject/ro-crate/blob/11efb935c1ebd51df594931664c154ea88cb2177/docs/1.2-DRAFT/crate-focus.md?plain=1#L119) additional schema.org vocab used the Schema.org convention of using rdfs:Class (rather than schema:Class) .

```
 {
      "@id": "txc:Annotation",
      "@type": "rdfs:Class",
      "name": "Annotation",
      "sameAs": "http://www.language-archives.org/REC/type-20020628.html#annotation",
      "rdfs:comment": "The resource includes information which annotates some other linguistic record.",
      "rdfs:label": "Annotation",
      "rdfs:subClassOf": {
        "@id": "schema:CreativeWork"
      }
    },

```

In the profile @stain you have done a few things I was not expecting:
-  Used schema:Class (implicitly via the use of Class as a @type)
-  Added DefinedTerm (eg for pcdm:Object)
- Omitted rdfs:label which is used in schema.org vocab

**URL**
[Profile RO-Crate Metadata Document](https://raw.githubusercontent.com/ResearchObject/ro-crate/master/docs/1.2-DRAFT/ro-crate-metadata.json)

**Suggested fix**

I would suggest that:
-  if something is a class it should be rdfs:Class as per the way Schema.org does it, or we change our advice about how to add extra Schema.org Style Schema terms.
- We do not add DefinedTerm as well as Class (I am writing a tool that can bring together the base Schema.org "schema" (classes and props) with RO-Crates extensions (like the PCDM terms and others) and I would like Classes and Properties to be as per Schema.org)
- We add rdfs:label as distinct from name (which may have spaces etc)

**Additional context**

**Comment by stain** [comment]
The reason for adding `DefinedTerm` to the `@type` array was to unify a bit with the ad-hoc classes as well as how terms are imported (which may also be used beyond Class/Property, e.g. for roles).  The idea being that any term imported is a `DefinedTerm` but could also be something else, in which case it should (at least) have the `name` from the Schema.org type.  

The move to then use schema.org's types `Class` and `Property` which mirrors `rdfs:Class` and `rdfs:Property` is perhaps more controversial as these are not commonly used for defining ontologies - as you point out not even used by Schema.org's own definitions, I also don't know of any tooling that support these. Perhaps they are sensible when only "quoting" such a class or property that is defined elsewhere, but not when defining classes/properties only directly within the profile.

In FAIR-IMPACT we have proposed to evolve the RO-Crate Profile to also work for any Semantic Artefact gathering - in which case it would IMHO stay in soft schema.org land because the artefacts would be fully defined in various ontology languages (e.g. SKOS, OWL2) and imposing `rdfs:*` would then be too strong.  @dgarijo may have different views.

So are you @ptsefton  then proposing to change/remove the new text in https://www.researchobject.org/ro-crate/1.2-DRAFT/appendix/jsonld.html#add-local-definitions-of-ad-hoc-terms (which suggests `rdfs:*` as an optional ad-on) from #232 to revert back to requiring `rdfs:*` as in RO-Crate 1.1?

**Comment by ptsefton** [comment]
@stian -- I'm not suggesting anything, still trying to get my head around this issue. At the moment we recommend additional Classes and Properties are use /rdfs?/ prefix and this is coded into the Javascript library and used for looking up term definitions [when they are present in the crate.](https://github.com/Arkisto-Platform/ro-crate-js/blob/e8fdd8705c04cbcb4c06ed641260a971824b66dc/lib/rocrate.js#L277). We are also maintaining a schema (vocabulary) for a major project using the Schema.org Style Schema method (SOSS) where the terms are `rdfs:Class`, `rdf:Property` and `(schema:)DefinedTerm` and `(schema):DefinedTermSet`. If we change to using (schema:)Class in profiles then I'd want to go back and change all that (and still support the use of `/rdfs?:/`)

I understand the idea of `DefinedTerm` but when we *know* something is a class or a property, isn't it better to bring it into the schema.org world by defining it in a way that tools CAN operate on it as part of a class hierarchy -- we are starting to build more of these in our work.

**Comment by stain** [comment]
Follow generally Crate-O's mode file to do Schema.org-style schemas: 

* `rdfs:Class` and `rdf:Property` with `domainIncludes` and `rangeIncludes`
* `DefinedTerm` for other terms/constants.

Avoid too many types.

**Comment by stain** [comment]
This is now implemented in #262 etc. as part of the revamp on Profiles.
